molmod/routes/main_routes.py

```diff
@@ -14,3 +24,53 @@ def index():
 def about():
     # abort(301)
     return render_template('about.html')
+
+
+@main_bp.route('/upload', methods=['GET', 'POST'])
+# Redirect user to Bioatlas CAS login
+@login_required
+def upload():
+    '''Checks whether logged-in user has required role for file upload.
+       Authorized users are sent to upload page; unauthorized users to custom
+       403 Forbidden page ('Data upload' menu item is actually hidden for those
+       - see __init.py__ + banner.html - but 403 is used if unauthorized user
+       types in '/upload' url).
+
+       Standard Bioatlas users have role 'ROLE_USER', and we require
+       'ROLE_COLLECTION_ADMIN' for upload, for now, but will have a specific
+       molmod-upload role added later.
+    '''
+
+    # Get CAS user roles from session
+    cas_attributes = session.get('CAS_ATTRIBUTES', None)
+    roles = cas_attributes['cas:authority'].split(',')
+    APP.logger.debug(
+        f'User has CAS roles: {roles} ')
+
+    # Stop unauthorized users
+    if os.getenv('UPLOAD_ROLE') not in roles:
+        abort(403)
+
+    # Enable authorized users
+    form = UploadForm()
+    upload_error = False
+    # If user has tried to submit file (POST)
+    if request.method == 'POST':
```

Comment:
This is fine (and I think some example code uses this style), but my preference is to avoid deep conditionals and rather bail out (e.g. return early).

Reply:
Learning as I go along, so please just edit where you think the code should be improved!